Backends/CLX/medium.lisp

```diff
@@ -124,19 +140,36 @@
         (unless (eq (line-style-joint-shape line-style)
                     (line-style-joint-shape old-line-style))
           (setf (xlib:gcontext-join-style gc)
-                (line-style-joint-shape line-style)))
+                (translate-join-shape (line-style-joint-shape line-style))))
         ;; we could do better here by comparing elements of the vector
         ;; -RS 2001-08-24
         (unless (and (eq (line-style-dashes line-style)
                          (line-style-dashes old-line-style))
                      (eq (line-style-unit line-style)
                          (line-style-unit old-line-style)))
-          (setf (xlib:gcontext-line-style gc)
-                (if (line-style-dashes line-style) :dash :solid)
-                (xlib:gcontext-dashes gc)
-                (case (line-style-dashes line-style)
-                  ((t nil) (round (* (line-style-scale line-style medium) 3)))
-                  (otherwise (line-style-effective-dashes line-style medium)))))))))
+          (let ((effective (line-style-effective-dashes line-style medium)))
+            (if-let ((dashes (line-style-dashes line-style)))
+              (setf (xlib:gcontext-line-style gc) :dash
+                    (xlib:gcontext-dashes gc) effective)
+              (setf (xlib:gcontext-line-style gc) :solid
+                    (xlib:gcontext-dashes gc) effective))))))))
```

Comment:
I suggest returning two values from `line-style-effective-dashes`: 1) `:dash` or `:solid` 2) scaled dash lengths.

Reply:
that would be fine but I want to factor out these functions (line-style-effective-dashes and line-style-effective-thickness), to be shared between backends, and these two values are very clx-specific.

Reply:
The function could return a Boolean as the second return value to indicate dashed vs. solid. That could be useful for other backends as well.

Also what is the meaning of `xlib:gcontext-dashes` when `xlib:gcontext-line-style` is `:solid`?

Reply:
as of their meaning – I don't know. I've decided to not change the behavior I do not understand (it was carried from the old code).

Reply:
I would skip the single-value behavior now, because we have similar functions in other parts of the codebase and it will be easier to change them at one go if they are consistent.